deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -226,365 +135,69 @@ def _execute_compaction(
-                        num_hash_groups=params.hash_group_count,
-                        object_store=params.object_store,
-                    ),
-                    write_to_partition=compacted_partition,
-                    compacted_file_content_type=params.compacted_file_content_type,
-                    primary_keys=params.primary_keys,
-                    sort_keys=params.sort_keys,
-                    merge_task_index=index,
-                    drop_duplicates=params.drop_duplicates,
-                    max_records_per_output_file=params.records_per_compacted_file,
-                    enable_profiler=params.enable_profiler,
-                    metrics_config=params.metrics_config,
-                    s3_table_writer_kwargs=params.s3_table_writer_kwargs,
-                    read_kwargs_provider=params.read_kwargs_provider,
-                    round_completion_info=round_completion_info,
-                    object_store=params.object_store,
-                    deltacat_storage=params.deltacat_storage,
-                    deltacat_storage_kwargs=params.deltacat_storage_kwargs,
-                    delete_strategy=delete_strategy,
-                    delete_file_envelopes=delete_file_envelopes,
-                    memory_logs_enabled=params.memory_logs_enabled,
-                    disable_copy_by_reference=params.disable_copy_by_reference,
-                )
-            }
-
-        merge_start = time.monotonic()
-        merge_tasks_pending = invoke_parallel(
-            items=all_hash_group_idx_to_obj_id.items(),
-            ray_task=mg.merge,
-            max_parallelism=task_max_parallelism,
-            options_provider=merge_options_provider,
-            kwargs_provider=merge_input_provider,
-        )
-        merge_invoke_end = time.monotonic()
-        logger.info(f"Getting {len(merge_tasks_pending)} merge results...")
-        merge_results: List[MergeResult] = ray.get(merge_tasks_pending)
-
-    logger.info(f"Got {len(merge_results)} merge results.")
-
-    merge_results_retrieved_at = time.time()
-    merge_end = time.monotonic()
-
-    total_dd_record_count = sum([ddr.deduped_record_count for ddr in merge_results])
-    total_deleted_record_count = sum(
-        [ddr.deleted_record_count for ddr in merge_results]
-    )
-    logger.info(
-        f"Deduped {total_dd_record_count} records and deleted {total_deleted_record_count} records..."
-    )
-
-    compaction_audit.set_input_records(total_input_records_count.item())
-
-    telemetry_time_merge = compaction_audit.save_step_stats(
-        CompactionSessionAuditInfo.MERGE_STEP_NAME,
+    build_uniform_deltas_result: tuple[
+        List[DeltaAnnotated], DeleteStrategy, List[DeleteFileEnvelope], Partition
+    ] = _build_uniform_deltas(
+        params, compaction_audit, input_deltas, delta_discovery_start
+    )
+    (
+        uniform_deltas,
+        delete_strategy,
+        delete_file_envelopes,
+        compacted_partition,
```

Comment:
Why is _build_uniform_deltas responsible for creating a compacted partition? I believe it's a separate concern.